pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -1101,9 +1101,13 @@ public CompletableFuture<Void> clearDelayedMessages() {
         if (delayedDeliveryTracker.isEmpty() && topic.getBrokerService()
                 .getDelayedDeliveryTrackerFactory() instanceof BucketDelayedDeliveryTrackerFactory) {
             synchronized (this) {
-                if (delayedDeliveryTracker.isEmpty()) {
-                    delayedDeliveryTracker = Optional
-                            .of(topic.getBrokerService().getDelayedDeliveryTrackerFactory().newTracker(this));
+                try {
+                    if (delayedDeliveryTracker.isEmpty()) {
+                        delayedDeliveryTracker = Optional
+                                .of(topic.getBrokerService().getDelayedDeliveryTrackerFactory().newTracker(this));
```

Comment:
@poorbarcode Because the dispatcher was closed before `unsubscribe` was called,  this lead to the tracker reference being lost, so we need to reinitialize the tracker.

Please see: https://github.com/apache/pulsar/blob/4f503fd886cd25670b6344defb8feb1ff2a6e816/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L1320-L1357

Reply:
> Because the dispatcher was closed before unsubscribe was called, this lead to the tracker reference being lost, so we need to reinitialize the tracker to clean up residual persistent data.

I see. Another question: the method `new tracker` will call `recover` internally, right? If yes, I feel it is an expensive task. Can we only set the Tracker to an unavailable state when closing the dispatcher to avoid recover the second time?

Reply:
You are right，I think we shoulud use another way to clean up data to avoid unnecessary recover，such as add a clean method in the factory.